jaxrs/WFLY-10361_EAP7-997_AsychClientHttpEngine.adoc

```diff
@@ -0,0 +1,59 @@
+= WFLY-10361 / EAP7-997 Async (NIO) request processing support in RESTEasy ClientHttpEngine
+:author:            Alessio Soldano
+:email:             asoldano@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+This is about providing an asynch version of the ClientHttpEngine abstraction within RESTEasy, to have non-blocking behavior on client side basically down up to the socket.
+A default implementation based on Apache HttpComponents HttpAsyncClient is also implemented.
```

Comment:
What about using the Undertow async HTTP client instead?

Reply:
Well, we can do that too (and possibly even change the default, if that's your point). Frankly speaking, this was included because RESTEasy users have been asking this for years and eventually even provided the changes to have it.